app/helpers/styled_form_builder.rb

```diff
@@ -59,6 +63,30 @@ def submit(value = nil, options = {})
     super(value, merged_options)
   end
 
+  def number_field(method, options = {})
+    base_classes = "form-field__input text-ellipsis text-subdued"
+    merged_options = options.merge(
+      class: [ base_classes, options[:class] ].compact.join(" ")
+    )
+
+    label = build_label(method, options)
+    field = super(method, merged_options)
+
+    build_styled_field(label, field, merged_options)
+  end
+
+  def date_field(method, options = {})
+    base_classes = "form-field__input text-ellipsis text-subdued"
+    merged_options = options.merge(
+      class: [ base_classes, options[:class] ].compact.join(" ")
+    )
+
+    label = build_label(method, options)
+    field = super(method, merged_options)
+
+    build_styled_field(label, field, merged_options)
+  end
```

Comment:
I think these can move up to the metaprogramming block we have here, as both number/date fields are captured there:

https://github.com/maybe-finance/maybe/blob/a7dfafc9072b2986834a0b9fff60e8241b1d82f1/app/helpers/styled_form_builder.rb#L6-L16